tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,91 @@
+package ru.sberbank.school.task02.calculator;
+
+import ru.sberbank.school.task02.ExtendedFxConversionService;
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.util.Beneficiary;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import java.math.BigDecimal;
+import java.util.*;
+import java.math.RoundingMode;
+
+public class ExtendedCurrencyCalculator extends CurrencyCalculator implements ExtendedFxConversionService {
+    public ExtendedCurrencyCalculator(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
+    }
+
+    private Optional<BigDecimal> extendedOperation(ClientOperation operation, Beneficiary beneficiary, List<Quote> quotes) {
+        Optional<Quote> quote = findQuote(new CompareQuotesBenificiary(operation, beneficiary), quotes, operation);
+        if (!quote.isPresent()) {
+            System.out.println("Return Optional.empty()");
+            return Optional.empty();
+        }
+        showQuote(quote.get());
+        if (operation == ClientOperation.SELL) {
+                return Optional.of(BigDecimal.valueOf(1).divide(quote.get().getBid(),10, RoundingMode.HALF_UP));
+        }
+        if (operation == ClientOperation.BUY) {
+            return Optional.of(BigDecimal.valueOf(1).divide(quote.get().getOffer(), 10, RoundingMode.HALF_UP));
+        }
+        System.out.println("Return Optional.empty()");
+        return Optional.empty();
+    }
+
+    Optional<Quote> findQuote(Comparator<Quote> comparator, List<Quote> quotes, ClientOperation operation) {
+        List<Quote> finalQuoteList = new ArrayList<>();
+        BigDecimal countCurr, volumeInCurr;
```

Comment:
Each variable declaration must be in its own statement.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1703379314>*